File: Chat-API/Chat-API/Models/Message.cs
namespace ChatWebSocketServer.Models
{
    public class Message
    {
        public string MessageId { get; set; }
        public Guid UserId { get; set; }
        public string Username { get; set; }
        public string Content { get; set; }
        public string Timestamp { get; set; }
        public MessageType Type { get; set; }


    }
    public class BaseMessage
    {
        public ActionType Action { get; set; }
        public object Data { get; set; }
    }

    public enum ActionType
    {
        Register = 1,
        Message = 2,
        SetStatus = 3
    }
}

File: chat-ui/src/types/Message.ts
export type BaseMessage = {
  Action: ActionType; // Action to perform (Register, Message, SetStatus)
  Data: object; // Payload data for the action
};

export enum ActionType {
  Register = 1, // Register a new user
  Message = 2, // Send or receive a message
  SetStatus = 3, // Update the status of a participant
}

export type Message = {
  MessageId: string; // Unique ID of the message
  UserId: string; // Unique ID of the sender
  Username: string | null; // Username of the sender
  Content: string; // Content of the message
  Type: MessageType; // Type of the message (Add, Edit, Delete)
  Timestamp?: string; // Optional timestamp for the message
};

export enum MessageType {
  Add = 1, // Add a new message
  Edit = 2, // Edit an existing message
  Delete = 3, // Delete a message
}